cmd/tasks_run.go

```diff
@@ -0,0 +1,242 @@
+	lagoonValues := lagoon.BuildValues{}
+	lYAML := lagoon.YAML{}
+	autogenRoutes := new(lagoon.RoutesV2)
+	mainRoutes := new(lagoon.RoutesV2)
+	activeStandbyRoutes := new(lagoon.RoutesV2)
+
+	err := collectBuildValues(true, &activeEnv, &standbyEnv, &lagoonEnvVars, &lagoonValues, &lYAML, autogenRoutes, mainRoutes, activeStandbyRoutes)
+	if err != nil {
+		return lagoon.YAML{}, nil, err
+	}
+
+	lagoonConditionalEvaluationEnvironment := tasklib.TaskEnvironment{}
+	if len(lagoonEnvVars) > 0 {
+		for _, envVar := range lagoonEnvVars {
+			lagoonConditionalEvaluationEnvironment[envVar.Name] = envVar.Value
+		}
+	}
+	return lYAML, lagoonConditionalEvaluationEnvironment, nil
+}
+
+func runTasks(taskType int, taskRunner iterateTaskFuncType, lYAML lagoon.YAML, lagoonConditionalEvaluationEnvironment tasklib.TaskEnvironment) error {
+
+	if namespace == "" {
+		//Try load from file
+		const filename = "/var/run/secrets/kubernetes.io/serviceaccount/namespace"
+		if _, err := os.Stat(filename); errors.Is(err, os.ErrNotExist) {
+			return fmt.Errorf("A target namespace is required to run pre/post-rollout tasks")
+		}
+		nsb, err := ioutil.ReadFile(filename)
+		if err != nil {
+			return err
+		}
+		namespace = strings.Trim(string(nsb), "\n ")
+	}
+
+	if taskType == PRE_ROLLOUT_TASKS {
+		fmt.Println("Executing Pre-rollout Tasks")
+		err2, done := taskRunner(lagoonConditionalEvaluationEnvironment, unwindTaskRun(lYAML.Tasks.Prerollout))
+		if done {
+			return err2
+		}
+		fmt.Println("Pre-rollout Tasks Complete")
+	} else {
+		fmt.Println("Skipping pre-rollout tasks")
+	}
+
+	if taskType == POST_ROLLOUT_TASKS {
+		fmt.Println("Executing Post-rollout Tasks")
+		fmt.Println(lYAML.Tasks.Postrollout)
+		err2, done := taskRunner(lagoonConditionalEvaluationEnvironment, unwindTaskRun(lYAML.Tasks.Postrollout))
+		if done {
+			return err2
+		}
+		fmt.Println("Post-rollout Tasks Complete")
+	} else {
+		fmt.Println("Skipping post-rollout tasks")
+	}
+	return nil
+}
+
+func unwindTaskRun(taskRun []lagoon.TaskRun) []lagoon.Task {
+	var tasks []lagoon.Task
+	for _, taskrun := range taskRun {
+		tasks = append(tasks, taskrun.Run)
+	}
+	return tasks
+}
+
+type iterateTaskFuncType func(tasklib.TaskEnvironment, []lagoon.Task) (error, bool)
+
+func iterateTasks(lagoonConditionalEvaluationEnvironment tasklib.TaskEnvironment, tasks []lagoon.Task) (error, bool) {
+	for _, task := range tasks {
+		runTask, err := evaluateWhenConditionsForTaskInEnvironment(lagoonConditionalEvaluationEnvironment, task)
+		if err != nil {
+			return err, true
+		}
+		if runTask {
+			err := runCleanTaskInEnvironment(task)
+			if err != nil {
+				return err, true
+			}
+			if err != nil {
+				return err, true
+			}
+		} else {
+			fmt.Printf("Conditional '%v' for task: \n '%v' \n evaluated to false, skipping\n", task.When, task.Command)
+		}
+	}
+	return nil, false
+}
+
+func getEnvironmentVariablesForConditionalEvaluation(pullWiderEnvironment bool) (tasklib.TaskEnvironment, error) {
```

Comment:
Is this used anymore? Can we trim it and the associated tests?

Reply:
Good catch. Removed :)